### PR TITLE
C3-509: Support for starting, stopping and resetting vm

### PR DESCRIPTION
**Pull request by avnish30jn**
**Jira Id**

C3-509

**Problem**

Support for starting, stopping and resetting vm is required

**Solution**

Added support for starting, stopping and resetting vm

**Unit testing**

Tried starting, stopping, and resetting vms. The action were successfully done.

### CODE REVIEW

File: virtualmachine/vsphere/util.go [hunk 1]
```diff
@@ -729,6 +729,31 @@ var start = func(vm *VM) error {
...
+		return err
+	}
+	vmMo, err := findVM(vm, dcMo, vm.Name)
+	if err != nil {
+		return err
+	}
+	vmo := object.NewVirtualMachine(vm.client.Client, vmMo.Reference())
+	resetTask, err := vmo.Reset(vm.ctx)
+	if err != nil {
+		return fmt.Errorf("error creating a reset task on the vm: %s", err)
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/util.go]: ->%v

File: virtualmachine/vsphere/vm.go [hunk 2]
```diff
@@ -175,9 +175,9 @@ type ClusterComputeResource struct {
 	FreeMemory     int64        `json:"free_memory"`
 	TotalStorage   int64        `json:"total_storage"`
 	FreeStorage    int64        `json:"free_storage"`
-	NoOfHosts      int          `json:"number_of_hosts"`
-	NoOfDatastores int          `json:"number_of_datastores"`
-	NoOfNetworks   int          `json:"number_of_networks"`
+	NoOfHosts      int          `json:"number_hosts"`
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/vm.go:178]: Are these fields not used so far? Do we require corresponding change at other places?

**Review by pankajtandon12 — CHANGES_REQUESTED** [review]

File: virtualmachine/vsphere/vm.go (hunk 2 above)
Comment (avnish30jn) [reply virtualmachine/vsphere/vm.go:178]: Yes these are not used, these will be consumed by controller when response is sent from c3ntry. Venkat suggested to remove of from these keys.

**Review by pankajtandon12 — APPROVED** [review]
LGTM

**Pull request merged**